src/builder/Accounts.sol

```diff
@@ -182,16 +181,17 @@ library Accounts {
         return findAssetPositions(assetAddress, chainAccounts.assetPositionsList);
     }
 
-    function findQuarkState(address account, Accounts.QuarkState[] memory quarkStates)
+    function findQuarkSecret(address account, Accounts.QuarkSecret[] memory quarkSecrets)
```

Comment:
Do we need to do it this way? Wouldn't just "next random" work?

Reply:
Synced offline on this. I don't think we resolved this discussion yet, but here's my response:

I considered specifying `nonceSecrets` on a non-account basis, but I decided to keep it account-based for a few reasons.

1. It conforms to the current interface that the client is passing in.
2. It reduces the complexity of the QuarkBuilder having to pick out `nonceSecrets` and keep track of which ones are already assigned to which operations.

Reply:
TBF, if we needed to do two operations on the same chain, this strategy wouldn't work? E.g. let's say we wanted to make two delayed operations on the same chain.

Reply:
If we have two operations on the same chain for the same account, we wrap those in a multi quark operation. So having a single `nonceSecret` for an account on each chain is guaranteed to be sufficient.